csi/src/node.rs

```diff
@@ -96,6 +96,32 @@ fn get_access_type(
     }
 }
 
+/// Detach the nexus device from the system, either at volume unstage,
+/// or after failed filesystem mount at volume stage.
+async fn detach(uuid: &Uuid, errheader: String) -> Result<(), Status> {
+    if let Some(device) = Device::lookup(uuid).await.map_err(|error| {
+        failure!(
```

Comment:
if we're idempotent shouldn't we return OK here?

Reply:
The code is a bit confusing but seems like in this case it would be an actual error and thus why we have Result<Option,> I guess...